app/src/main/java/com/example/todo/SplashScreenActivity.java

```diff
@@ -27,13 +24,56 @@ protected void onCreate(Bundle savedInstanceState) {
         Notification notification = new Notification(getApplicationContext());
         notification.createNotificationsChanel();
 
-        lunchMainActivity();
+        lunchActivity();
+    }
+
+    private boolean connected() {
+        ConnectivityManager connectivityManager = (ConnectivityManager) getSystemService(Context.CONNECTIVITY_SERVICE);
+        NetworkInfo activeNetwork = connectivityManager.getActiveNetworkInfo();
+        return activeNetwork != null && activeNetwork.isConnectedOrConnecting();
+    }
+
+    private String getUserID() {
+        SharedPreferences sharedPreferences = getSharedPreferences("user_data", MODE_PRIVATE);
+
+        return sharedPreferences.getString("user_id", null);
     }
 
-    private void lunchMainActivity() {
-        Intent intent = new Intent(this, MainActivity.class);
-        intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TASK | Intent.FLAG_ACTIVITY_NEW_TASK);
-        startActivity(intent);
+    private void lunchActivity() {
+        Intent intent;
+        if (connected()) {
+            if (getUserID() != null)
+                intent = new Intent(this, MainActivity.class);
+            else
+                intent = new Intent(this, LoginActivity.class);
+
+            intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TASK | Intent.FLAG_ACTIVITY_NEW_TASK);
+            startActivity(intent);
+        } else {
+            showAlertDialog();
+        }
+
     }
 
+    private void showAlertDialog() {
+        AlertDialog.Builder builder = new AlertDialog.Builder(this)
+                .setTitle("Needed internet connection")
+                .setMessage("Please, check your internet connection")
+                .setPositiveButton("Try again", new DialogInterface.OnClickListener() {
+                    @Override
+                    public void onClick(DialogInterface dialog, int which) {
+                        lunchActivity();
```

Comment:
very bad method. separate method for checking internet and separate method for lunch activity